docs/api.yaml

```diff
@@ -0,0 +1,88 @@
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: CartBooking API
+  license:
+    name: GPLv3
+servers:
+  - url: https://api.cartbooking.com/
+paths:
+  /bookings:
```

Comment:
I am not sure how to expose things for the first part.
<img width="281" alt="screen shot 2017-11-18 at 10 29 03 am" src="https://user-images.githubusercontent.com/1817347/32973402-55dc2714-cc4b-11e7-9023-c20d5035941c.png">

To fill that part, the calendar, I have a couple of ideas. Open to better recommendations though. We can either expose a collection of bookings and pass a query filter `?filter=publisher_id` but it feels incorrect.

Maybe a better approach is to `GET /publishers/{publisherId}/bookings` in which case we would be exposing the collection of bookings that belong to a given publisher. I think is a bit better idea. We can also add some filtering to get only the "current month". Something like `?filter=encoded([month=yyyy/dd,...])` Not sure if is a lot better. I think it can work, but maybe there is something cleaner to expose that.

What would you recommend @johnrivelt?

Reply:
Hi, @serroba ,

I presume roles like the following
```
- ROLE_SUPER_ADMIN (technician who setup this app)
  Super is super
- ROLE_ADMIN (an overseer)
  CRUD locations (private)
  CRUD users (private)
  CRUD shifts (private)
  CRUD bookings (private)
  R bookings (public)

- ROLE_USER (a publisher)
  CRUD bookings (private)
  R bookings (public)
  R locations (inherit from own overseer)
  R shifts (inherit from own overseer)
```

Can you implement API style like as you said ? 
It looks good to me.
```
/api/bookings/?{filter}   #{filter} is only for GET method
/api/bookings/{id}?{filter} ...   [GET, POST, PUT, DELETE]
... etc
```

*Regarding `GET /publishers/{publisherId}/bookings`, 
1. For ROLE_USER, whatever Token or login auth  we use, 
a user is already authenticated and will have a session. 
In that case, `/api/bookings ` should be okay.

2. For ROLE_ADMIN, how about ?
```
GET /api/publishers/ or
GET /api/publishers/all/bookings
    [
        { id:1, name:'serroba', bookings:[
          {booking_id:1, location_id:1, shift_id:1, ....}
        ]},
        { id:2, name:'johnc', bookings:[
         {}
        ]},
    ]
GET /api/publishers/{publisherId} or
GET /api/publishers/{publisherId}/bookings
        { id:1, name:'serroba', bookings:[
          {booking_id:1, location_id:1, shift_id:1, ....}
        ]}
```

Please use which one is easier for you.

Reply:
Thanks @johnrivelt 
Give me some time to think about the options.

Which one is the first one you would like to have exposed, so you can use it ? Like, if you need to think about what comes first to unblock you. What would you need first ?

Reply:
hey @johnrivelt how would you like to filter by date?
If I make the bookings a subresource of a publisher, I guess you will want to be able to get only the bookings of the current month. A collection of bookings for the month right ?

Would you advice something different? I can give you the entire collection but that will end up being a massive JSON